src/book/mod.rs

```diff
@@ -270,8 +271,15 @@ impl MDBook {
                     _ => continue,
                 };
 
-                let path = self.source_dir().join(&chapter_path);
-                info!("Testing file: {:?}", path);
+                if chapter.is_some()
+                    && ch.name != chapter.unwrap()
+                    && chapter_path.to_str().unwrap() != chapter.unwrap()
```

Comment:
I'd prefer to avoid using unwrap here.  Perhaps more like this:

```suggestion
                if let Some(chapter) = chapter {
                    if ch.name != chapter && chapter_path.to_str() != Some(chapter) {
```

However, this logic seems wrong. It seems to require that the `--chapter` value passed in equals *both* the chapter name and its filename. I would expect it to match one or the other. Perhaps something more like?

```rust
!(ch.name == chapter || chapter_path.to_str() == Some(chapter))
```

Reply:
it works, notice this is the inverse of the test `!(ch.name == chapter || chapter_path == chapter)`